test/e2e/seeder/contract-address-registry.js

```diff
@@ -12,16 +12,39 @@ class ContractAddressRegistry {
    * @param contractAddress
    */
   storeNewContractAddress(contractName, contractAddress) {
-    this.#addresses[contractName] = contractAddress;
+    if (!this.#addresses[contractName]) {
+      this.#addresses[contractName] = [];
+    }
+    this.#addresses[contractName].push(contractAddress);
   }
 
   /**
    * Get deployed contract address by its name (key).
+   * Returns the most recently deployed contract of this type.
    *
    * @param contractName
    */
   getContractAddress(contractName) {
-    return this.#addresses[contractName];
+    const addresses = this.#addresses[contractName];
+    return addresses && addresses.length > 0
+      ? addresses[addresses.length - 1]
+      : undefined;
+  }
+
+  /**
+   * Get all deployed contract addresses in deployment order.
+   *
+   * @returns Array of all deployed contract addresses in deployment order
+   */
+  getAllDeployedContractAddresses() {
```

Comment:
:information_source:  new method, so we can work with multiple deployed contracts, not just 1 (or 1 by 1)